src/extensions/agent_modules/shared/state_store/src/agent_state_store.c

```diff
@@ -8,19 +8,21 @@
 #include "aduc/agent_state_store.h"
 #include <aduc/logging.h>
 #include <aduc/string_c_utils.h> // IsNullOrEmpty
-#include <parson_json_utils.h>
 #include <semaphore.h>
+#include <stdbool.h> // bool
 #include <stdlib.h>
 #include <string.h>
-#include <stdbool.h> // bool
-
-// Internal state data
-static JSON_Value* inmem_state_data = NULL;
 
-static JSON_Value* durable_state_data = NULL;
+#define ADUC_DEFAULT_DEVICE_REGISTRATION_STATE_POLL_INTERVAL_SECONDS (10)
 
+// Internal state data
 static bool isInitialized = false;
 
+// MQTT SPEC says topics can have max len of 65536 utf-8 encoded bytes.
+//
+// TODO: Figure out a lower, tighter maximum based on topic template path segment maximums, or if this maximum is not sufficient.
+#define MAX_ADU_MQTT_TOPIC_BYTE_LEN 2048
+
```

Comment:
The technical max number of utf-8 bytes for topics (according to MQTT spec) is 65536 bytes, so I've increased it to that so that we do not artificially preclude future scenarios.

I've also updated `MAX_ADU_DEVICE_ID_BYTE_LEN` and `MAX_ADU_SCOPE_ID_BYTE_LEN` based on tracking those down.